executor/analyze.go

```diff
@@ -135,6 +161,34 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	if err != nil {
 		return err
 	}
+	if needGlobalStats {
+		for _, info := range globalStatsMap {
+			globalStats, succ := statsHandle.MergePartitionStats2GlobalStats(infoschema.GetInfoSchema(e.ctx), info.tableID, info.isIndex, info.idxID)
+			if succ {
+				for i := 0; i < globalStats.Num; i++ {
+					hg := globalStats.Hg[i]
+					cms := globalStats.Cms[i]
+					topN := globalStats.TopN[i]
+					if hg == nil || cms == nil || topN == nil {
+						succ = false
+						break
+					}
+					err = statsHandle.SaveStatsToStorage(info.tableID, globalStats.Count, info.isIndex, hg, cms, topN, info.statsVersion, 1)
```

Comment:
Should we guarantee that all data is committed in one transaction?

Reply:
If it is not done in one transaction, after we save partition-level stats to storage, another `analyze` may update the stats of the same partition, and we read new version of partition-level stats for merging. In the case `info.statsVersion` may be outdated. Is it ok for `analyze`?

Reply:
After thinking it again, it's OK for me that these glotbal-stats are not updated in one transaction, so the case you said is also OK for me... @xuyifangreeneyes

Reply:
I see. It seems not be a problem.